File: 0226-invert-binary-tree/0226-invert-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
     //levelwise invert the tree
        if(!root) return {};
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
          TreeNode* node = q.front();
          q.pop();
          if(node->left) q.push(node->left);
          if(node->right) q.push(node->right);
          TreeNode* temp = node->left;
            node->left = node->right;
            node->right = temp;
        }
        return root;
    }
};


// easy peasy bfs gotcha

File: 0226-invert-binary-tree/0226-invert-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

interface TreeNode{
    val:number;
    left:TreeNode | null;
    right:TreeNode | null;
}

function invertTree(root: TreeNode | null): TreeNode | null {
   if(!root) return null; 
    const q: TreeNode[] = [];
    q.push(root);
    while(q.length>0){
        const node = q.shift() as TreeNode;
        if(node.left){
            q.push(node.left);
        }
        if(node.right){
            q.push(node.right);
        }
        const temp = node.left;
        node.left = node.right;
        node.right = temp;
    }
    return root;
};





